remix-resolve/src/resolve.ts

```diff
@@ -0,0 +1,149 @@
+import axios, { AxiosResponse } from 'axios'
+
+import { Api, ModuleProfile, API } from 'remix-plugin'
+
+export interface RemixResolve extends Api {
+  type: 'remix-resolve'
+  events: {}
+  resolve: Function
+}
+
+export const RemixResolveProfile: ModuleProfile<RemixResolve> = {
+  type: 'remix-resolve',
+  methods: ['resolve']
+}
+
+
+export interface Imported {
+  content: string;
+  cleanURL: string;
+  type: string;
+}
+
+interface PreviouslyHandledImports {
+  [filePath: string]: Imported
+}
+
+interface Handler {
+  type: string;
+  match(url: string): any;
+  handle(match: any): any;
+}
+
+export class RemixResolveApi implements API<RemixResolve> {
```

Comment:
I think we should not make the implementation directly depends on remix-plugin.

here a proposal, let's discuss on it:
  - remix-* codes (like remix-resolve, remix-debug, ...) will not depend on remix-plugin in a first place, which means that it does not have to implement `API<... Something ...>` . the resolve.ts of this PR remains the exact same but without `implements API<RemixResolve>`.
 - in the same module, we declare another `package.json` with name `remix-resolve-plugin` which then have remix-plugin as dependency, and which return `RemixResolveProfile` and the api.

that would also make the unit testing easier, cause they are at the moment depending on remix-plugin.

Reply:
So we will have two `package.json` for the same module ? Not sure how does that work. But I agree that all these modules should not directly depend on `remix-plugin` API.

Reply:
> So we will have two package.json for the same module

yes and with different name, the other package.json will just depend on remix-plugin and do the mapping of the API.
one other solution (might be easier), would be to have one npm module (package.json) only dedicated on providing remix-plugin API of core remix-* module. basically this npm module would depends on remix-* libraries and on remix-plugin and its task is to remix-plugin(ed) remix-* libraries.

Reply:
shall we merge the current code now? just need to remove `implements API<RemixResolve>` and remix-plugin reference;